kernel/task.h

```diff
@@ -188,4 +194,26 @@ extern void (*exit_hook)(struct task *task, int code);
 // Will ensure that the -pid part always fits, then will fit as much of comm as possible.
 void update_thread_name(void);
 
+// To collect statics on which tasks are blocked we need to proccess areas
+// of code which could block our task (e.g reads or writes). Before executing
+// of functions which can block the task, we mark our task as blocked and
+// unblock it after the function is executed.
+__attribute__((always_inline)) inline int enter_blocking_area(void) {
+    lock(&pids_lock);
+    current->io_block = 1;
+    blocked_task_count++;
+    unlock(&pids_lock);
+    return 0;
+}
+
+__attribute__((always_inline)) inline int leave_blocking_area(void) {
+    lock(&pids_lock);
+    current->io_block = 0;
+    blocked_task_count--;
+    unlock(&pids_lock);
+    return 0;
+}
+
+#define TASK_BLOCKING_AREA for(int i = enter_blocking_area(); i < 1; leave_blocking_area(), i++)
```

Comment:
What do you think of naming this `TASK_MAY_BLOCK`? Then `enter_blocking_area`/`leave_blocking_area` becomes `task_may_block_start()` and `task_may_block_end()`